### DISCUSSION TITLE
3.2.2に関する不具合や質問はこちら

### DISCUSSION THREAD

**Discussion by dongaba** [dongaba/TVerRec#112] [Issues]
### 3.2.2 Changed

- シリーズタイトルの取得先を変更 (理由は、Series.Content.Titleだとファイル名が冗長になることがあることと、複数シーズン配信時に最新シーズン名になってしまうことがあるため)
- 番組移動時に、mp4とtsファイルだけでなくすべてのファイルを移動の対象にするように変更

### 3.2.2 Fixed

- ダウンロード先ディレクトリ、作業ディレクトリ、移動先ディレクトリの指定時に、末尾に`\`や`/`が入っていてもエラーとならないように修正
- 番組移動時に特殊文字を含むフォルダの取り扱いができていなかった問題を修正

## What's Changed
* Dev (#110) by @dongaba in https://github.com/dongaba/TVerRec/pull/111

**Full Changelog**: https://github.com/dongaba/TVerRec/compare/v3.2.1...v3.2.2

<hr /><em>This discussion was created from the release <a href='https://github.com/dongaba/TVerRec/releases/tag/v3.2.2'>3.2.2</a>.</em>

**Comment by kokoritsu** [comment]
dongabaさん、TVerRecいつも便利に使わさせていただいてます。ありがとうございます。
Windows11 24H2、PowerShell 7.4.6の環境でWindows CUI版を利用しています。

3.2.2からファイルの移動の際にエラーが出るようになってしまいました。
move_video.ps1の143行目で「Cannot create 'example' because a file or directory with the same name already exists.」が出ていて、ファイルの移動が出来ないようです。

自分の環境では143行目を修正して動かしていますが、他の環境での動作確認は出来なので、ご報告です。

```
- try { $null = Move-Item -LiteralPath ('{0}' -f $moveFromPath) -Destination $(Split-Path $moveToPath -Parent) -Force }
+ try { $null = Get-ChildItem -Path $moveFromPath -File | Move-Item -Destination $moveToPath -Force}
```

**Reply by dherhi** [reply]
私も同環境で同じエラーがでたのでkokoritsuさんの修正を導入しましたところ動作致しました｡
また､`「ダブルチート 偽りの警官 Season2」WOWOW×テレビ東京 共同製作連続ドラマ` のような「」を含むフォルダが適切に移動されませんでした｡

**Reply by dongaba** [reply]
ご報告ありがとうございます、確認します。

@dherhi さん
「」を含むフォルダが移動されない件ですが、 @kokoritsu さんの修正を行った後も移動されない、という理解であっていますか?

**Reply by dongaba** [reply]
私の環境では再現しないのですが、いただいたコードをもとに修正を行い、v3.2.3としてbetaブランチにマージしました。
v3.2.3では他の機能追加も予定しているので、それらの目処がついたらmasterにマージしてリリースします。

**Reply by kokoritsu** [reply]
@dongabaさん
自分の環境では「」を含むフォルダの移動に問題はありません。参考にしてください。
